src/datatypes/interval.coffee

```diff
@@ -207,6 +207,18 @@ module.exports.Interval = class Interval
     else
       @.start().sameAs(other.start(), precision) and @.end().sameAs(other.end(), precision)
 
+  sameOrBefore: (other, precision) ->
+    if !@end()? || !other.start()?
+      return null
+    else
+      return @end().sameOrBefore(other.start(), precision)
```

Comment:
This function (and `sameOrAfter`) assumes that `other` is not `null`.  In the current code, that's a safe assumption since it is only called in one place and that already checks for `null` -- but I think it would be nice to check `!other?` here as well in case we use this method elsewhere in the future and don't do the null check.

Reply:
As for priority: I'd like to see this change before I approve the PR.

Reply:
Addressed and tests added in 0a304c492fab291d54c2abdd9b02d0a3086f7677.